jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -523,40 +525,40 @@ default List<Resource> split(String str, String delims, boolean unwrap)
+                    // or "C:/path/to/dir/*" (windows java Path syntax)
+                    // or "C:\path\to\dir\*" (windows native syntax)
+                    // or even a URI like "file:///path/to/dir/*" (always an absolute URI)
+                    // and "file:///C:/path/to/dir/*"
+                    // and "jar:file:///C:/path/to/foo.jar!/deep/*"
+                    String rawDir = reference.substring(0, reference.length() - 2);
+                    // Convert to a URI without loading it as a Resource (yet).
+                    URI uri = URIUtil.toURI(rawDir);
+                    // Load rawDir as a Resource
+                    Resource dir = newResource(uri);
                     if (dir.isDirectory())
                     {
+                        // Loop through resource entries for content that will match glob.
                         List<Resource> expanded = dir.list();
                         expanded.sort(ResourceCollators.byName(true));
                         expanded.stream().filter(r -> FileID.isLibArchive(r.getName())).forEach(list::add);
                     }
                 }
                 else
                 {
-                    // Simple reference. Could have a scheme like jar:file:, or just file:.
-                    // Or it could be a relative reference, in which case we need to
-                    // ensure it is absolute. Otherwise, comparisons between Resources
-                    // that point to the same resource but one is relative and one is absolute
-                    // will fail.
-                    if (URIUtil.hasScheme(reference))
+                    // Simple reference, could be relative, could be windows syntax, could be a URI.
+                    URI uri = URIUtil.toURI(reference);
+                    if ("jar".equals(uri.getScheme()) && unwrap)
                     {
-                        // Could be a jar:file url, ensure it is unwrapped back to the file
-                        // otherwise it will be a MountedPathResource and consume a mount point
-                        // that might be unnecessary - the caller should always decide whether to mount
-                        URI uri = new URI(reference);
-                        if (unwrap)
-                            list.add(newResource(URIUtil.unwrapContainer(uri)));
-                        else
-                            list.add(newResource(uri.toASCIIString()));
+                        list.add(newResource(URIUtil.unwrapContainer(uri)));
                     }
                     else
                     {
-                        Path p = Paths.get(reference);
-                        if (!p.isAbsolute() && LOG.isDebugEnabled())
-                            LOG.warn("Non-absolute path: {}", reference);
-                        list.add(newResource(p.toAbsolutePath()));
+                        list.add(newResource(uri));
```

Comment:
I'm a bit cautious that this is a behaviour change - previously relative references were ignored, but now they will be used.

Reply:
Relative references are 100% necessary to support here.
This is a code path for many configuration from jetty-home/jetty-base and it's properties.
It's very common to use relative paths in the configuration, so that the same configuration can be supported across dev, qa, docker images, production, etc...

Reply:
Prior to your change, relative references were not ignored.
Admittedly they were poorly tested, but that lack of testing is hopefully addressed in this PR.

Reply:
Example:

```
[joakim@hyperion jetty-home-12.0.10]$ jshell --class-path lib/jetty-util-12.0.10.jar:lib/logging/jetty-slf4j-impl-12.0.10.jar:lib/logging/slf4j-api-2.0.12.jar 
|  Welcome to JShell -- Version 17.0.15
|  For an introduction type: /help intro

jshell> import org.eclipse.jetty.util.resource.*;

jshell> try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable()) {
   ...>     var result = resourceFactory.split("bar/a,bar/b,foo/z");
   ...>     System.out.println(result);
   ...> }
[file:///home/joakim/code/jetty/distros/jetty-home-12.0.10/bar/a, file:///home/joakim/code/jetty/distros/jetty-home-12.0.10/bar/b, file:///home/joakim/code/jetty/distros/jetty-home-12.0.10/foo/z]

jshell> /exit
|  Goodbye
```